api/src/user/controllers/role.controller.ts

```diff
@@ -147,12 +153,26 @@ export class RoleController extends BaseController<
   @CsrfCheck(true)
   @Delete(':id')
   @HttpCode(204)
-  async deleteOne(@Param('id') id: string) {
-    const result = await this.roleService.deleteOne(id);
-    if (result.deletedCount === 0) {
-      this.logger.warn(`Unable to delete Role by id ${id}`);
-      throw new NotFoundException(`Role with ID ${id} not found`);
+  async deleteOne(@Param('id') id: string, @Req() req: Request) {
+    const userRoles = (req.user as User).roles;
+
+    const associatedUser = await this.userService.findOne({
+      roles: { $in: [id] },
+    });
+    if (userRoles.includes(id)) {
+      throw new ForbiddenException("Your account's role can't be deleted");
+    } else if (associatedUser) {
+      throw new ForbiddenException('Role is associated with other users');
+    } else {
+      try {
+        const result = await this.roleService.deleteOne(id);
+        if (result.deletedCount === 0) {
+          throw new NotFoundException(`Role with ID ${id} not found`);
+        }
+        return result;
+      } catch (error) {
+        throw new NotFoundException(`Role with ID ${id} not found`);
+      }
```

Comment:
```suggestion
        const result = await this.roleService.deleteOne(id);
        if (result.deletedCount === 0) {
          throw new NotFoundException(`Role with ID ${id} not found`);
        }
        return result;
```
The throw new `NotFoundException` is redundant, and the `try-catch` block isn’t necessary here. If a different error occurs, returning a `NotFoundException` could be misleading, as we should return an `InternalServerErrorException` or another appropriate error instead.